File: 525-contiguous-array/contiguous-array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        count = 0
        lookup = {0: -1}
        output = 0

        for i in range(len(nums)):
            if (nums[i] == 0):
                count += 1
            else:
                count -= 1

            if (count in lookup):
                output = max(output, i - lookup[count])
            
            else:
                lookup[count] = i

        return output
        

File: 525-contiguous-array/contiguous-array.js
var findMaxLength = function(nums) {
    let count = 0
    let maxLength = 0
    let lookup = {0: -1}

    for (let i = 0; i < nums.length; i++) {
        let num = nums[i]
        if (num == 0) count -= 1
        else if (num == 1) count += 1
        if (!(count in lookup)) lookup[count] = i
        else {
            let length = i - lookup[count]
            maxLength = Math.max(maxLength, length)
        }
    }   
    return maxLength

};